sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -2889,5 +2889,129 @@ def math_pipeline():
                 add(nums=dsl.Collected(t.output))
 
 
+class TestValidIgnoreUpstreamTaskSyntax(unittest.TestCase):
+
+    def test_basic_permitted(self):
+
+        @dsl.component
+        def fail_op(message: str) -> str:
+            """Fails."""
+            import sys
+            print(message)
+            sys.exit(1)
+            return message
+
+        @dsl.component
+        def print_op(message: str = 'default'):
+            """Prints a message."""
```

Comment:
nit: can remove the docstrings unless they are there to help explain the behavior being tested